saleor/order/utils.py

```diff
@@ -977,13 +890,4 @@ def remove_discount_from_order_line(
         ]
     )
 
-    update_taxes_for_order_line(order_line, order, manager, tax_included)
-    order_line.save(
-        update_fields=[
-            "unit_price_gross_amount",
-            "unit_price_net_amount",
-            "total_price_net_amount",
-            "total_price_gross_amount",
-            "tax_rate",
-        ]
-    )
+    fetch_order_prices_if_expired(order, manager)
```

Comment:
All calculations of new prices should go thourhg the fetch_order_prices (see lines above 880-891). Also `fetch_order_prices_if_expired` has been build to delay a price calculation, and call it on the resolver and palces where we really need a new prcies. It is not 1:1 a replacment for `update_order_prices` method, so we can't replace one with the other. If we change something in order, we reset an expiration time.